cpp/src/arrow/type_traits.h

```diff
@@ -614,9 +637,28 @@ using is_string_type =
 template <typename T, typename R = void>
 using enable_if_string = enable_if_t<is_string_type<T>::value, R>;
 
+template <typename T>
+using is_binary_view_like_type = std::is_base_of<BinaryViewType, T>;
```

Comment:
Is this really necessary used anywhere?

Reply:
I use it in a couple places when I need to specifically target string_view and not string

https://github.com/apache/arrow/blob/868858f5d104a3e8e1cec8d7278ed88bb1f8bf3c/cpp/src/arrow/array/array_binary_test.cc#L562-L566